backend/server.js

```diff
@@ -206,6 +223,31 @@ sbp('okTurtles.data/set', PUBSUB_INSTANCE, createServer(hapi.listener, {
       socket.send(createNotification(NOTIFICATION_TYPE.VERSION_INFO, versionInfo))
     }
   },
+  socketHandlers: {
+    close () {
+      const socket = this
+      const { server } = this
+
+      const subscriptionId = socket.pushSubscriptionId
+
+      if (!subscriptionId) return
+      delete socket.pushSubscriptionId
+
+      if (!server.pushSubscriptions[subscriptionId]) return
+
+      server.pushSubscriptions[subscriptionId].sockets.delete(socket)
+      delete socket.pushSubscriptionId
+
+      if (server.pushSubscriptions[subscriptionId].sockets.size === 0) {
+        server.pushSubscriptions[subscriptionId].subscriptions.forEach((channelID) => {
+          if (!server.subscribersByChannelID[channelID]) {
+            server.subscribersByChannelID[channelID] = new Set()
+          }
+          server.subscribersByChannelID[channelID].add(server.pushSubscriptions[subscriptionId])
+        })
+      }
```

Comment:
What does this `close()` handler do and why is it not contain some of the same code from `[NOTIFICATION_TYPE.UNSUB]`, like `deleteChannelFromSubscription`?

(Maybe a comment explaining this?)

Reply:
`close()` is an entirely different thing from `UNSUB`. `close()`, which is called when the WS is closed, tells the server to send subsequent messages to that client as push notifications.

`UNSUB` isn't substantially different from function than what we have on `master`, except that it'll now also remove the channels from the push subscription, so that when `close()` is used, no push notifications are sent for UNSUBed channels.